### PR TITLE
Add MQTT demo dependencies target for FreeRTOS console support

### PR DESCRIPTION
**Pull request by aggarw13**
### Background

With the new _dependency-free_ architecture of the `coreMQTT` library, the new MQTT demos have multiple dependencies (on libraries like `retry_utils`, `secure_sockets`, etc). This is different from the old dependency model of demos where they carried a single dependency on their matching library (like `demo_defender` depending on `AFR::defender`), and the matching library bringing all the indirect dependencies (like `AFR::defender` dependning on (old) `AFR::mqtt`).  The new dependency model affects the FreeRTOS console, as the FreeRTOS console depends on library targets brining in all the indirect dependencies. 

### Fix
Rhis PR adds a single dependency wrapper module for the `coreMQTT` library named `core_mqtt_demo_dependencies` that represents the list of ALL dependencies for the new MQTT demos within a single module target. This will allow the the demo target to depend on a single module target to continue supporting the FreeRTOS console experience.

* Module name: `core_mqtt` - Purely, the library of the _dependency-free_ MQTT library
* Module name: `core_mqtt_demo_dependencies` - The metadata module target that represents collection of ALL modules that the demo depends on which include **core_mqtt**, **logging**, **transport_interface_secure_sockets**, **retry_utils** and **secure_sockets**

### CODE REVIEW

**Review by abhidixi11 — APPROVED** [review]

**Review by abhidixi11 — DISMISSED** [review]

**Review by abhidixi11 — APPROVED** [review]

**Review by tgsong — DISMISSED** [review]

File: libraries/core_mqtt_demo_dependencies.cmake [hunk 1]
```diff
@@ -1,6 +1,6 @@
 afr_module(NAME core_mqtt )
 
-afr_set_lib_metadata(ID "core_mqtt")
+afr_set_lib_metadata(ID "core_mqtt_demo_dependencies")
```
Comment (aggarg) [inline libraries/core_mqtt_demo_dependencies.cmake:3]: This should be core_mqtt?

File: libraries/core_mqtt_demo_dependencies.cmake [hunk 2]
```diff
@@ -29,7 +29,8 @@ afr_module_sources(
         ${MQTT_SERIALIZER_SOURCES}
         # List of files added to the target so that these are available
         # in code downloaded from the FreeRTOS console.
-        ${CMAKE_CURRENT_LIST_DIR}/core_mqtt.cmake
+        ${CMAKE_CURRENT_LIST_DIR}/coreMQTT/mqttFilePaths.cmake
+        ${CMAKE_CURRENT_LIST_DIR}/core_mqtt_demo_dependencies.cmake
```
Comment (aggarg) [inline libraries/core_mqtt_demo_dependencies.cmake:33]: Shouldn't `core_mqtt_demo_dependencies.cmake` be part of `core_mqtt_demo_dependencies` module?

**Review by aggarg — APPROVED** [review]

File: libraries/core_mqtt_demo_dependencies.cmake (hunk 2 above)
Comment (aggarw13) [reply libraries/core_mqtt_demo_dependencies.cmake]: Yes, moved it toward the bottom of the file for better readability. 
To the metadata logic, the location of the metadata functions does not matter. The matching with the module is done based on the name of the file (thus, kept the file name as `core_mqtt_demo_dependencies.cmake`)

File: libraries/core_mqtt_demo_dependencies.cmake (hunk 1 above)
Comment (aggarw13) [reply libraries/core_mqtt_demo_dependencies.cmake]: We will not be adding any metadata for `core_mqtt` but only for the dependencies target

**Review by abhidixi11 — APPROVED** [review]

**Pull request merged**